plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -972,10 +972,15 @@ public void dropTable(String databaseName, String tableName, boolean deleteData)
         }
     }
 
-    private static void deleteDirRecursive(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path)
+    private static void deleteDirRecursive(Optional<ConnectorIdentity> hiveIdentity, HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path)
     {
         try {
-            hdfsEnvironment.getFileSystem(context, path).delete(path, true);
+            if (hiveIdentity.isPresent()) {
```

Comment:
Since `hdfsContext` is only used here, we can remove it and change the code to something like
```java
hdfsEnvironment.getFileSystem(identity.orElse(DEFAULT_METASTORE_USER), path).delete(path, true);
```
No need to call `hdfsEnvironment.getConfiguration()` explicitly since `getFileSystem()` does that when you only specify a path.

Reply:
make sense

Reply:
actually, there are two signatures for `getFileSystem`, one is `public FileSystem getFileSystem(HdfsContext context, Path path)`, the other `public FileSystem getFileSystem(ConnectorIdentity identity, Path path, Configuration configuration)`. The only way to get `Configuration` is through `HdfsContext` and `Path`. 

In other words, we can't get the file system with only the identity and the path.

Reply:
Ah, makes sense. We could do
```java
HdfsContext context = new HdfsContext(identity.orElseGet(() -> ConnectorIdentity.ofUser(DEFAULT_METASTORE_USER)));
hdfsEnvironment.getFileSystem(context, path).delete(path, true);

Reply:
makes sense. I'll remove the member variable as well since we are recreating the context here